api/src/test/java/org/openmrs/module/adminui/utils/RegexUtilTest.java

```diff
@@ -0,0 +1,57 @@
+package org.openmrs.module.adminui.utils;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class RegexUtilTest {
+
+	@Test
+	public void shouldEvaluateToFalseForStringWithNoDigit(){
+		RegexUtil util = new RegexUtil("thisissomething", RegexUtil.MUST_CONTAIN_DIGIT_REGEX);
+		Assert.assertFalse("String should have at least a digit", util.matcher());
+	}
+
+	@Test
+	public void shouldEvaluateToTrueForStringWithDigits(){
+		RegexUtil util = new RegexUtil("thisis12something", RegexUtil.MUST_CONTAIN_DIGIT_REGEX);
+		Assert.assertTrue("Could not match string with at least a digit", util.matcher());
+	}
+
+	@Test
+	public void shouldEvaluateToTrueForStringWithDigitsAndSpecialChars(){
+		RegexUtil util = new RegexUtil("thisis12something@#", RegexUtil.MUST_CONTAIN_DIGIT_REGEX);
+		Assert.assertTrue("Could not match string with a digit and special chars", util.matcher());
+	}
+
+	@Test
+	public void shouldEvaluateToFalseForStringWithoutNoneDigit(){
+		RegexUtil util = new RegexUtil("12345678", RegexUtil.MUST_CONTAIN_NONE_DIGIT_REGEX);
+		Assert.assertFalse("String should have at least a none-digit", util.matcher());
+	}
+
+	@Test
+	public void shouldEvaluateToTrueForStringWithAtLeastANoneDigit(){
+		RegexUtil util = new RegexUtil("12345678thisfollowsdigits", RegexUtil.MUST_CONTAIN_NONE_DIGIT_REGEX);
+		Assert.assertTrue("Should pass for string with a none-digit", util.matcher());
```

Comment:
You also need a test where the string contains special characters only or digits mixed with special characters